packages/fxa-auth-server/scripts/cleanup-partial-firestore-customers.ts

```diff
@@ -0,0 +1,199 @@
+    this.accountManager = Container.get(AccountManager);
+  }
+
+  /**
+   * Get customers from Firestore paginated by batchSize
+   * @returns A list of customers from firestore
+   */
+  private async *getFirestoreCustomersUids() {
+    this.log.debug(
+      'CleanupFirestoreHelper.getFirestoreCustomersUids.start',
+      {}
+    );
+    const customerCollection = `${this.config.authFirestore.prefix}stripe-customers`;
+    let customerSnap = await this.firestore
+      .collectionGroup(customerCollection)
+      .orderBy(FieldPath.documentId())
+      .limit(this.batchSize)
+      .get();
+    while (!customerSnap.empty) {
+      yield customerSnap.docs.map((doc) => doc.id);
+      customerSnap = await this.firestore
+        .collectionGroup(customerCollection)
+        .orderBy(FieldPath.documentId())
+        .startAfter(customerSnap.docs.at(-1))
+        .limit(this.batchSize)
+        .get();
+    }
+  }
+
+  /**
+   * Determines which uids should be deleted
+   * @param uids Expects an array of account uids
+   * @returns A list of uids to be deleted
+   */
+  private async getUidsForDelete(uids: string[]) {
+    this.log.debug('CleanupFirestoreHelper.getUidsForDelete.start', {
+      nrOfUids: uids.length,
+    });
+
+    // Get all accounts for uids
+    const accounts = await this.accountManager.getAccounts(uids);
+    const accountUids = accounts.map((account) =>
+      uuidTransformer.from(account.uid)
+    );
+    // Filter uids where account does not exist
+    // Uids where account doesn't exist, should be deleted
+    const uidsForDelete = uids.filter(
+      (uid) => !accountUids.find((accountId) => accountId === uid)
+    );
+
+    return uidsForDelete;
+  }
+
+  /**
+   * Create Cloud Task to delete account
+   * @param uids Expects an array of account uids
+   */
+  private async callPartialDelete(uids: string[]) {
+    this.log.debug('CleanupFirestoreHelper.callPartialDelete.start', {
+      uidsForDelete: uids.length,
+    });
+    if (this.dryRun) {
+      this.log.debug('CleanupFirestoreHelper.callPartialDelete.skipped', {
+        dryRun: this.dryRun,
+        uidNumbers: uids.length,
+      });
+
+      return;
+    }
+
+    await Promise.all(
+      uids.map((uid) =>
+        this.accountDeleteManager.enqueue({
+          uid,
+          reason: ReasonForDeletionOptions.Cleanup,
+        })
+      )
+    );
+  }
+
+  async cleanup() {
+    this.log.debug('CleanupFirestoreHelper.cleanup.start', {
+      batchSize: this.batchSize,
+      dryRun: this.dryRun,
+    });
+    let batchCounter = 0;
+    for await (const uids of this.getFirestoreCustomersUids()) {
+      this.log.info('CleanupFirestoreHelper.cleanup', {
+        msg: `Batch ${batchCounter * this.batchSize} - ${
+          (batchCounter + 1) * this.batchSize - 1
+        }`,
+      });
+      // 2. Find uids that need to be deleted
+      const uidsForDelete = await this.getUidsForDelete(uids);
```

Comment:
One of the suggestions I liked from Refactoring code is to avoid comments by naming the variables or functions sufficiently descriptive that they aren't necessary. Perhaps:
```typescript
const uidsToDelete = await this.filterUidsForDeletion(uids);
```

Reply:
Or if this is doing a specific job, indicate that, long function names are fine.
```typescript
const uidsToDelete = await this.filterUidsNotInDatabase(uids);
```

Reply:
Ah shucks. Missed the follow up comment. My bad. I like the change, so can fix it up on my next PR.